### ISSUE TITLE
Wrong TransferFunction from arx() if nb > na

### ISSUE THREAD

**Issue by mapi1** [baggepinnen/ControlSystemIdentification.jl#57]
I was trying to fit some all zero tfs using `arx()` and the first obstacle I came across was the restriction `na >= 1`, which does not seem necessary as `getARXregressor()` seems to handle the case `na = 0` well. 
I did some further experimenting and found, that for the case `na < nb` a wrong tf gets returned. A minimal example would be sth. like:
```
N = 512
G =  tf([1, -2, 3], [1, -0.5, 0,0], 1)
u = randn(N)
y = lsim(G, u, 1:N)[1][:]
d = iddata(y, u, 1)
Gest = arx(d, 1, 3)
```  
Here the parameters are correctly identified, but the order of z is off. The error, as far as I can tell, comes from `params2poly()`, where some zeros should be padded to a, so the correct tf is recovered. I would like to do a PR fixing this, but I wanted to get to you first making sure this is not an error on my side misusing the package/function. I also remember that some versions ago the restrictions were even stricter, only allowing `all(nb .<= na)`, maybe that was the reason. 

Beside that let me say thank you for the great package you created here, which I really heavily utilize for my work and which has served me really well so far!

**Comment by baggepinnen** [comment]
Thanks for your kind words, and for opening the issue! It actually brings up a number of problems
- An incorrect tf should obviously no be returned under any circumstance.
- Estimation of a model with only zeros is actually handled by `impulseest`, but this might not be discoverable to users who do not think about the connection between an infinite-order FIR model and a moving average model. 
- `arx` might be a misleading function name for a model that does not contain `y` in the regressor, it will not in fact be autoregressive then. Nevertheless, it would be good to support models where there are more coefficients in the numerator than the denominator, where `na=1` would be the extreme case. A PR making sure this works well would be greatly appreciated. Perhaps the correct way of handling it would be to specify an additional delay to the numerator, i.e., an extra factor `z^(-d)` that shifts the input in time to make the system proper. This would make sure the order of the polynomials are correct, or error if the delay is misspecified, resulting in a non-proper transfer function. The specification of a delay would be useful also for models that actually has an input delay. Does this sound reasonable?

**Comment by mapi1** [comment]
Thanks for your fast replay!

Regarding your points:

- I was not aware of this function, though it makes sense.
- Indeed, for `na = 1` the name `arx` would be misleading. But for the testing I did so far `na = 0` worked fine, so why not allow it as an extreme case? This is also how the matlab `arx` handles this, so it would bring consistency for users coming from there. 

How would you add an factor `z^(-d)` to the numerator? What I would do at the moment would be sth. like `a = [a; zeros(d)]` in `params2poly()`, but maybe their is a better/ intended way to do this? 
With your last sentence are you referring to adding an additional argument `nk` to the function for arbitrary input-output delay? Like `arx(d::AbstractIdData, na, nb, nk = zeros(size(nb)); λ = 0, estimator = \, stochastic = false)` orienting on the matlab names again and unsure if it should be an optional or keyword argument.

I would prepare a PR now, so there is a more solid ground for a discussion, which will be necessary as this would be my first PR :sweat_smile:

**Comment by baggepinnen** [comment]
`a = [a; zeros(d)]` seems like the only way it can be done at the moment, our discrete-time model structures do not support a delay in any other way.

Yep, an additional parameter would be the way to go, just like you suggested. I haven't used matlab in almost a decade so I can not remember exactly how they handle things, but providing a vector of delays that defaults to `zeros(size(nb))` seems reasonable. I think having it a keyword argument would be a better approach as it is not obvious to me what a fourth argument to this function would be. I am also not a big fan of the name `nk`, it's not nearly as well established as `A,B` for the polynomials of the tf. `inputdelay` or something like that seems more self explaining.

**Comment by baggepinnen** [comment]
Closed in #58
